openedx/features/redhouse_panel/api/v0/serializers.py

```diff
@@ -9,7 +21,98 @@ class SiteSerializer(serializers.Serializer):
     address = serializers.CharField()
 
 
-class UserSerializer(serializers.ModelSerializer):
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ('name', 'year_of_birth')
+
+
+class UserAccountSerializer(serializers.ModelSerializer):
+    profile = UserProfileSerializer()
+    can_access_panel = serializers.SerializerMethodField('has_panel_permission')
+    is_instructor = serializers.SerializerMethodField('can_create_courses')
+
     class Meta:
         model = User
-        fields = ('id', 'first_name', 'last_name', 'is_active')
+        read_only_fields = ('id',)
+        fields = ('username', 'email', 'is_active', 'profile', 'can_access_panel', 'is_instructor')
+
+    def has_panel_permission(self, obj):
+        if obj.groups.filter(name__in=[REDHOUSE_PANEL_GROUP_NAME]).exists():
+            return True
+        return False
+
+    def can_create_courses(self, obj):
+        request = self.context['request']
+        edly_user_info_cookie = request.COOKIES.get(settings.EDLY_USER_INFO_COOKIE_NAME, None) if request else None
+        edx_org = get_edx_org_from_cookie(edly_user_info_cookie)
+
+        if auth.user_has_role(obj, GlobalCourseCreatorRole(edx_org)):
+            return True
+        return False
+
+    def create(self, validated_data):
+        # create user
+        profile_data = validated_data.pop('profile')
+        can_access_panel = self.initial_data.get('can_access_panel', None)
```

Comment:
can we avoid using `initial_data`?

Reply:
@mubtadaali 
The other way to get the data is by request `request.data.get('can_access_panel')`. Do you recommend that?

Reply:
we can design our serializer to behave differently for reading and writing. This way we will be able to use validated_data instead of initial_data.

Reply:
I don't see any issue with the `initial_data`, so using this for now. Let's see in the future if we face any problem.

Reply:
Using initial data means we are bypassing validations. If we are expecting a boolean and someone sends a string as the input to your serializer then it could break your code.

Either we should validate values that are directly used from initial_data by ourselves or we should design serializer in such a way it automatically validates those fields.

Reply:
@Luqman-Ud-Din Done. I have added validation for extra fields.

Reply:
👍